### PR TITLE
Compress CI artifacts

### PR DESCRIPTION
**Pull request by hennevogel**
Use less of our CI credits on storage by compressing logs and
capybara dumps. Also makes downloading things faster...

### CODE REVIEW

**Comment by hennevogel** [comment]
Screenshot from 2022-07-28 11-54-50

https://app.circleci.com/pipelines/github/openSUSE/open-build-service/7470/workflows/0d96b07c-ac0a-47e7-8c12-e9751c71480a/jobs/90696/artifacts

File: .circleci/conditional_config.yml [hunk 1]
```diff
@@ -251,8 +244,12 @@ jobs:
           root: .
           paths:
             - src/api/coverage_results
-      - <<: *store_rspec_artifacts
-      - <<: *store_rspec_test_results
+      - store_test_results:
+          path: /home/frontend/rspec
+      - run: *compress_log_artifacts
+      - store_artifacts:
```
Comment (dmarcoux) [inline .circleci/conditional_config.yml:250]: This step is always happening after `compress_log_artifacts`. Could we also extract this like you did with `compress_log_artifacts` and re-use this code wherever needed?

File: .circleci/conditional_config.yml [hunk 2]
```diff
@@ -397,12 +402,16 @@ jobs:
...
-          destination: feature
-      - <<: *store_capybara_artifacts
       - store_test_results:
           path: /home/frontend/feature
+      - run: *compress_log_artifacts
+      - store_artifacts:
+          path: ./src/api/log
+          destination: log
+      - run: *compress_capybara_artifacts
+      - store_artifacts:
```
Comment (dmarcoux) [inline .circleci/conditional_config.yml:412]: This step is always happening after `compress_capybara_artifacts`. Could we also extract this like you did with `compress_capybara_artifacts` and re-use this code wherever needed?

File: .circleci/conditional_config.yml (hunk 1 above)
Comment (hennevogel) [reply .circleci/conditional_config.yml:250]: Yeah we could but I' not so sure about the `- <<:` syntax...

Comment (dmarcoux) [reply .circleci/conditional_config.yml:250]: Alright, it can also stay as it is now.

**Review by dmarcoux — APPROVED** [review]

File: .circleci/conditional_config.yml (hunk 1 above)
Comment (hennevogel) [reply .circleci/conditional_config.yml:250]: The problem is that the anchor then doesn't have to reference some simple mapping but an array which needs to get merged with another array. I'm not getting any wiser from the yaml docu...

**Comment by hennevogel** [comment]
coveralls hangs. codecov is fine. merging!

**Pull request merged**